index.ts

```diff
@@ -64,17 +64,32 @@ async function run(): Promise<void> {
     console.log('You can configure the deploy branch by setting the `deploy-branch` input for this action.')
 
     await exec.exec(`git init`, [], {cwd: `${workingDir}/public`})
-    await exec.exec(`git config user.name`, [github.context.actor], {
-      cwd: `${workingDir}/public`,
-    })
-    await exec.exec(`git config user.email`, [`${github.context.actor}@users.noreply.github.com`], {
+    await exec.exec(`git config user.name`, [core.getInput('git-config-name') || github.context.actor], {
```

Comment:
To make it a bit easier to parse, let's split up getting the name into a `const` similar to some of the other options: 

```suggestion
    const gitUserName = core.getInput('git-config-name') || github.context.actor
    await exec.exec(`git config user.name`, [gitUserName], {
```